ChopChop/Storage/Models/Ingredients/QuantityRecord.swift

```diff
@@ -1,23 +1,25 @@
 enum QuantityRecord: Equatable {
     case count(Double)
-    case mass(Double)
-    case volume(Double)
+    case mass(Double, unit: MassUnit)
+    case volume(Double, unit: VolumeUnit)
 }
 
 extension QuantityRecord: Codable {
     enum CodingKeys: CodingKey {
-        case count, mass, volume
+        case count, mass, volume, unit
     }
 
     init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
 
         if let value = try container.decodeIfPresent(Double.self, forKey: .count) {
             self = .count(value)
-        } else if let value = try container.decodeIfPresent(Double.self, forKey: .mass) {
-            self = .mass(value)
-        } else if let value = try container.decodeIfPresent(Double.self, forKey: .volume) {
-            self = .volume(value)
+        } else if let value = try container.decodeIfPresent(Double.self, forKey: .mass),
+                  let unit = try container.decodeIfPresent(MassUnit.self, forKey: .unit) {
+            self = .mass(value, unit: unit)
+        } else if let value = try container.decodeIfPresent(Double.self, forKey: .volume),
+                  let unit = try container.decodeIfPresent(VolumeUnit.self, forKey: .unit) {
+            self = .volume(value, unit: unit)
```

Comment:
Has this been confirmed to be working properly with the database (both the encoding and decoding)?

Reply:
I think this requires some updating of the database to also store the unit of the quantity.